pkg/controller/operators/catalog/operator.go

```diff
@@ -1105,13 +1117,16 @@ func (o *Operator) ensureSubscriptionInstallPlanState(logger *logrus.Entry, sub
 	out.Status.InstallPlanRef = ref
 	out.Status.Install = v1alpha1.NewInstallPlanReference(ref)
 	out.Status.State = v1alpha1.SubscriptionStateUpgradePending
+	if failForwardEnabled && ip.Status.Phase == v1alpha1.InstallPlanPhaseFailed {
+		out.Status.State = v1alpha1.SubscriptionStateFailed
```

Comment:
I'm slightly concerned that `UpgradeFailed` may be misleading given the initial installation is not really an upgrade. However, given there is no other status, this is the only choice.

Reply:
On another note, the subscription status doesn't really matter given the CSV's status is what we care about.

Reply:
>I'm slightly concerned that UpgradeFailed may be misleading given the initial installation is not really an upgrade. However, given there is no other status, this is the only choice.

In the context of "Failing Forward", I believe that placing the subscription in the failed state is fitting as the subscription kicked off an upgrade and the InstallPlan failed.